libflux/go/libflux/bench_test.go

```diff
@@ -0,0 +1,111 @@
+// +build libflux
+
+package libflux_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/influxdata/flux/ast"
+	"github.com/influxdata/flux/libflux/go/libflux"
+	"github.com/influxdata/flux/parser"
+)
+
+func BenchmarkRustParse(b *testing.B) {
+	var fluxFile string
+	func() {
+		f, err := os.Open("./testdata/bench.flux")
+		if err != nil {
+			b.Fatalf("could not open testdata file: %v", err)
+		}
+		defer func() {
+			_ = f.Close()
+		}()
+		bs, err := ioutil.ReadAll(f)
+		fluxFile = string(bs)
+	}()
+
+	bcs := []struct {
+		name string
+		fn   func(string) error
+	}{
+		{
+			name: "rust parse and return handle",
+			fn:   ParseReturnHandle,
+		},
+		{
+			name: "rust parse and return JSON",
+			fn:   ParseReturnJSON,
+		},
+		{
+			name: "rust parse and deserialize JSON",
+			fn:   ParseAndDeserializeJSON,
+		},
+		{
+			name: "rust parse and deserialize flatbuffer",
+			fn:   ParseAndDeserializeFB,
+		},
+		{
+			name: "go parse",
+			fn:   GoParse,
+		},
+	}
+	for _, bc := range bcs {
+		bc := bc
+		if success := b.Run(bc.name, func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
+				if err := bc.fn(fluxFile); err != nil {
+					b.Fatal(err)
+				}
+			}
+		}); !success {
+			b.Fatalf("benchmark %q failed", bc.name)
+		}
+	}
+}
+
+func ParseReturnHandle(fluxFile string) error {
+	 p := libflux.Parse(fluxFile)
+	 p.Free()
+	return nil
+}
+
+func ParseReturnJSON(fluxFile string) error {
+	p := libflux.Parse(fluxFile)
+	defer p.Free()
+	_, err := p.MarshalJSON()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func ParseAndDeserializeJSON(fluxFile string) error {
+	p := libflux.Parse(fluxFile)
+	defer p.Free()
+	bs, err := p.MarshalJSON()
+	if err != nil {
+		return err
+	}
+	var bb = bytes.NewBuffer(bs)
+	d := json.NewDecoder(bb)
+	pkg := &ast.File{}
+	if err := d.Decode(pkg); err != nil {
+		return err
+	}
+	return nil
+}
+
+func ParseAndDeserializeFB(fluxFile string) error {
+	_ = libflux.ParseIntoFbs(fluxFile)
+	return nil
+}
+
+
+func GoParse(fluxFile string) error {
+	_ = parser.ParseSource(fluxFile)
```

Comment:
This might not test the go parser. I think this uses the rust parser when `libflux` is used as a tag. I think you can fix this by switching the import to `internal/parser` instead.

Reply:
Yes, you were exactly right about this.